math/approximate_pi.cpp

```diff
@@ -0,0 +1,75 @@
+/**
+ * @file
+ * @brief Implementation to  calculate an estimate of the number π (Pi).
+ * https://en.wikipedia.org/wiki/File:Pi_30K.gif
+ *
+ * @details
+ * We take a random point P with coordinates (x, y) such that 0 ≤ x ≤ 1 and 0 ≤ y ≤ 1. If x² + y² ≤ 1, then the 
+ * point is inside the quarter disk of radius 1, otherwise the point is outside.
+ * We know that the probability of the point being inside the quarter disk is equal to π/4
+ * double approx(vector<Point> &pts) which will use the points pts (drawn at random) to 
+ * return an estimate of the number π
+ * \note This implementation is better than naive recursive or iterative
+ * approach.
+ *
+ * @author [Qannaf AL-SAHMI](https://github.com/Qannaf)
+ */
+
+#include <iostream>  /// for IO operations
+#include <vector>    /// for std::vector
+#include <cstdlib>  /// for rand
+
+/**
+ * @namespace math
+ * @brief Mathematical algorithms
+ */
+namespace math {
+/**
+ * @brief Function to test above algorithm
+ * @param n description
+ * @returns void
+ */
+
+    /**
+     * structure of points containing two numbers, respectively x and y such that 0 ≤ x ≤ 1 and 0 ≤ y ≤ 1. 
+    */
+    typedef struct {
+    double x;
+    double y;
+    } Point;
+
+    double  approximate_pi(const std::vector<Point> &pts) {
+    /**
+     * This function use the points pts (drawn at random) to return an estimate of the number π  using the given points
+     * @param pts Each item of pts contains a point. A point is represented by a structure containing exactly 
+     * two numbers, respectively x and y such that 0 ≤ x ≤ 1 and 0 ≤ y ≤ 1. 
+     * pts always contains at least one item
+     * @return  an estimate of the number π
+     */
+        {
+            int count =0;    // Points in cercle
+            for(Point p:pts)
+                if(p.x * p.x + p.y*p.y <= 1)
+                    ++count;
+            
+            return 4.0*count/pts.size();
+        }
+    }
+}  // namespace math
+
```

Comment:
Please add self-test cases so we can verify your program works properly (**see the [typical structure of a program](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#typical-structure-of-a-program) as a reference**). Let us know if you need any help here or in our [**Discord server**](https://discord.gg/c7MnfGFGa6). 😃